src/normalizer.ts

```diff
@@ -105,6 +107,38 @@ function normalizeProp(p: Parameter | Property | XtpItemType, s: Schema) {
   }
 }
 
+function validateArrayItems(arrayItem: XtpItemType | parser.XtpItemType | undefined, location: string): void {
+  if (!arrayItem || !arrayItem.type) {
+    return;
+  }
+
+  validateTypeAndFormat(arrayItem.type, arrayItem.format, location);
+}
+
+function validateTypeAndFormat(type: XtpType, format: XtpFormat | undefined, location: string): void {
+  const validTypes = ['string', 'number', 'integer', 'boolean', 'object', 'array'];
+  if (!validTypes.includes(type)) {
+    throw new ValidationError(`Invalid type ${type}`, location);
```

Comment:
Maybe something like:

```suggestion
    throw new ValidationError(`Invalid type ${type}.`, `Options are: ${validTypes.join(', ')}`, location);
```